gremlinapi/oauth.py

```diff
@@ -23,49 +27,285 @@
 
 from http.client import HTTPResponse
```

Comment:
Exposing http.client outside of the API call class should be avoided. Ideally the API client class should not be returning anything but objects we create.